operators.py

```diff
@@ -8,504 +8,550 @@
-                    auxX=cfg.OVERLAPX*i+x+cfg.RANGEX[0]+cfg.TESSELLATIONDISPLACEMENT[0]
-
-                    if cfg.GRAPHTYPE=="CYCLE":
-                        if auxX>cfg.RANGEX[1]:
-                            auxX=cfg.RANGEX[0]
-                        index=(auxX)
-                    elif cfg.GRAPHTYPE=="LINE":
-                        if auxX>cfg.RANGEX[1]:
-                            auxX=cfg.RANGEX[0]
-                        index=(auxX-cfg.RANGEX[0])
-
-                    array_Index[counter]=index
-                    counter=counter+1                    
-            counter=0        
-            for k in range(0,cfg.STAGGERED_COEFICIENTS.shape[1],2):
-                array_Values[counter]=cfg.STAGGERED_COEFICIENTS[1][k]+1J*cfg.STAGGERED_COEFICIENTS[1][k+1]
-                counter=counter+1                    
 
-            for i1 in range(totalVerticesPerPatch):
-                for j1 in range(totalVerticesPerPatch):
-                    value=2*array_Values[i1]*np.conj(array_Values[j1])
-                    if (array_Index[i1]==array_Index[j1]):
-                        value=value-1
-                    f.write("%d %d %1.16f %1.16f\n"%(array_Index[i1]+1,array_Index[j1]+1,value.real,value.imag))
+    totalVerticesPerPatch = cfg.TESSELLATIONPOLYGONS[0]
+
+    array_Index = np.zeros((totalVerticesPerPatch, 1), dtype=int)
+    array_Values = np.zeros((totalVerticesPerPatch, 1), dtype=complex)
+
+    #    cfg.Uodd=np.zeros((cfg.STATESIZE,cfg.STATESIZE),dtype=float)
+
+    counter = 0
+
+    #    print cfg.TESSELLATIONDISPLACEMENT
+
+    for i in range(int(
+            cfg.TOTAL_PATCHES_IN_X)):  ### Run in all patches in axis X
+        counter = 0
+        for x in range(int(cfg.TESSELLATIONPOLYGONS[0])
+                       ):  ### Run in all vertices per patches in axis X
+            auxX = cfg.OVERLAPX * i + x + cfg.RANGEX[
+                0] + cfg.TESSELLATIONDISPLACEMENT[0]
+
+            if cfg.GRAPHTYPE == "CYCLE":
+                if auxX > cfg.RANGEX[1]:
+                    auxX = cfg.RANGEX[0]
+                index = (auxX)
+            elif cfg.GRAPHTYPE == "LINE":
+                if auxX > cfg.RANGEX[1]:
+                    auxX = cfg.RANGEX[0]
+                index = (auxX - cfg.RANGEX[0])
+
+            array_Index[counter] = index
+            counter = counter + 1
+        counter = 0
+        for k in range(0, cfg.STAGGERED_COEFICIENTS.shape[1], 2):
+            array_Values[counter] = cfg.STAGGERED_COEFICIENTS[1][
+                k] + 1J * cfg.STAGGERED_COEFICIENTS[1][k + 1]
+            counter = counter + 1
+
+        for i1 in range(totalVerticesPerPatch):
+            for j1 in range(totalVerticesPerPatch):
+                value = 2 * array_Values[i1] * np.conj(array_Values[j1])
+                if (array_Index[i1] == array_Index[j1]):
+                    value = value - 1
+                f.write("%d %d %1.16f %1.16f\n" %
```

Comment:
Is the indentation difference here on purpose? The file write is moved out of the if block